### PR TITLE
Go publish improvements

### PR DESCRIPTION
**Pull request by QuintinWillison**
I saw an opportunity for us to check the bulk of the Go code generation stuff for every pull request raised, so here it is.

I've split _build_ and _publish_ apart, linked by the new `generated` folder.

### CODE REVIEW

**Review by tomkirbygreen — APPROVED** [review]
Makes sense.

File: scripts/publish-go.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 #
-# A script to generate language specific code from source protocol files and
-# publish them to downstream GitHub repositories.
+# A script to publish generate language specific code to downstream GitHub repositories.
```
Comment (lmars) [inline scripts/publish-go.sh:3]: This sentence now doesn't seem to make sense, should it perhaps be:

"A script to publish generated code to downstream GitHub repositories."

**Review by lmars — APPROVED** [review]
LGTM other than minor comment

File: scripts/publish-go.sh (hunk 1 above)
Comment (QuintinWillison) [reply scripts/publish-go.sh]: Thanks. Done: https://github.com/ably/ably-common/pull/90/commits/6bdafd8a7dbc9ae950517cfbc45ac1f562d522e3

I'm going to take the liberty of resolving this conversation on your behalf so I can get this pull request landed.

**Pull request merged**